### PR TITLE
fix version checks for Intel MPI, fix check for local usage, change mpirun check using $EBROOT* and $EBVERSION*, avoid use of --setmpi in end2end tests

### PR DESCRIPTION
**Pull request by Caylo**
wrt #114

### CODE REVIEW

File: lib/vsc/mympirun/mpi/mpi.py [hunk 1]
```diff
@@ -847,7 +864,7 @@ def make_mpiexec_hydra_options(self):
             else:
                 self.log.raiseException("There is no launcher specified, and no default launcher found")
 
-        if launcher != 'local':
+        if not self.isLocal():
```
Comment (boegel) [inline lib/vsc/mympirun/mpi/mpi.py]: we don't use `CamelCase` for method names, should be `is_local` or `islocal`

File: lib/vsc/mympirun/mpi/mpi.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def what_mpi(name):
     # mympirun was not called through a known alias, so find out which MPI
     # implementation the user has installed
     for mpi in supp_mpi_impl:
-        if mpi._is_mpirun_for(mpirun_path):
+        if mpi._is_mpirun_for():
```
Comment (boegel) [inline lib/vsc/mympirun/mpi/mpi.py]: hmm, this looks strange now... maybe we should change the method name too?

File: lib/vsc/mympirun/mpi/mpi.py [hunk 3]
```diff
@@ -318,32 +335,26 @@ def __init__(self, options, cmdargs, **kwargs):
...
+            elif vs:
+                LOGGER.DEBUG("found EBVERSION%s: %s" % (cls._mpirun_for.upper(), vs))
+                version = cls._mpirun_version(vs)
             else:
-                # do version check (reg_match.group(1) is the version number)
-                LOGGER.debug("checking if mpirun version is equal or greater than required")
-                return cls._mpirun_version(reg_match.group(1))
-        else:
-            return False
+                LOGGER.debug("environment variable EBVERSION%s not found, skipping version check" % cls._mpirun_for)
```
Comment (boegel) [inline lib/vsc/mympirun/mpi/mpi.py]: missing `upper()` -> construct the env var name once above, reuse it

```python
env_var_name = 'EBVERSION' + cls._mpirun_for.upper()
```

**Comment by Caylo** [comment]
todo: remove `--setmpi impirun` in all end2end tests (fails atm, find out why)

File: lib/vsc/mympirun/mpi/intelmpi.py [hunk 4]
```diff
@@ -252,5 +251,6 @@ class IntelHydraMPIPbsdsh(IntelHydraMPI):
     """ MPI class for IntelMPI, with hydra and supporting pbsdsh """
     # pbsdsh is supported from Intel MPI 5.0.3:
     # https://software.intel.com/sites/default/files/managed/b7/99/intelmpi-5.0-update3-releasenotes-linux.pdf
-    _mpirun_version = staticmethod(lambda x: LooseVersion(x) >= LooseVersion("5.0.3"))
+    _mpirun_version = staticmethod(lambda ver: version_in_range(ver, '5.0.3', None))
+    HYDRA = True
```
Comment (boegel) [inline lib/vsc/mympirun/mpi/intelmpi.py]: this should not be needed?

File: test/end2end.py [hunk 5]
```diff
@@ -250,37 +248,44 @@ def test_option_multi(self):
...
+
+    def test_option_universe_hydra(self):
+        """Test --universe command line option with hydra impi"""
+        install_fake_mpirun('mpirun', self.tmpdir, 'impi', '5.124.67')
+
+        cmd = "%s %s --universe 1 hostname"
         ec, out = run_simple(cmd % (sys.executable, self.mympiscript))
-        self.assertTrue(np_regex.search(out))
-        self.assertFalse(ncpus_regex.search(out))
+        print out
```
Comment (boegel) [inline test/end2end.py]: this should be here anymore :)

**Comment by boegel** [comment]
@Caylo remarks fixed + `_mpirun_for` extended to also check location of `mpirun` command

**Comment by boegel** [comment]
lgtm, thanks @Caylo!

**Pull request merged**